patricia_trie/src/triedb.rs

```diff
@@ -365,27 +365,30 @@ impl<'a, H: Hasher, C: NodeCodec<H>> Iterator for TrieDBIterator<'a, H, C> {
 								let l = self.key_nibbles.len();
 								self.key_nibbles.truncate(l - n.len());
 							},
-							OwnedNode::Branch(_, _) => { self.key_nibbles.pop(); },
+							OwnedNode::Branch(_) => { self.key_nibbles.pop(); },
 							_ => {}
 						}
 						IterStep::PopTrail
 					},
-					(Status::At, &OwnedNode::Leaf(_, ref v)) | (Status::At, &OwnedNode::Branch(_, Some(ref v))) => {
+					(Status::At, &OwnedNode::Branch(ref branch)) if branch.has_value() => {
+						return Some(Ok((self.key(), DBValue::from_slice(branch.get_value().unwrap()))));
```

Comment:
Instead of scary-looking `unwrap()` can you use `expect("with the proof that this is ok, e.g. 'already checked that it has_value()'; qed")`?

Reply:
sure